.github/workflows/test.yml

```diff
@@ -12,16 +12,31 @@ jobs:
       uses: actions/checkout@v2
       with:
         fetch-depth: 0
-    - name: setup python
-      uses: actions/setup-python@v4
+    - name: Cache conda
+      uses: actions/cache@v2
+      env:
+        # Increase this value to reset cache if etc/example-environment.yml has not changed
+        CACHE_NUMBER: 0
       with:
-        cache: pip
-        cache-dependency-path: pyproject.toml
-        python-version: "${{ matrix.python-version}}"
-    - name: upgrade dependencies
-      run: |-
-        python -m pip install --upgrade pip hatch
-        pip install -e.[test]
+        path: ~/conda_pkgs_dir
+        key:
+          ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+          hashFiles('test-environment.yml') }}
+    - uses: conda-incubator/setup-miniconda@v2

```

Comment:
As a greenfield project, you could also consider the smaller-scope https://github.com/mamba-org/setup-micromamba